sdk/core/azure_core_amqp/src/error/mod.rs

```diff
@@ -0,0 +1,288 @@
+            | AmqpErrorKind::LinkStateError(e)
+            | AmqpErrorKind::ConnectionDropped(e)
+            | AmqpErrorKind::TransferLimitExceeded(e)
+            | AmqpErrorKind::FramingError(e)
+            | AmqpErrorKind::IdleTimeoutElapsed(e) => Some(e.as_ref()),
+            AmqpErrorKind::ManagementStatusCode(_, _)
+            | AmqpErrorKind::NonTerminalDeliveryState
+            | AmqpErrorKind::SimpleMessage(_)
+            | AmqpErrorKind::IllegalDeliveryState
+            | AmqpErrorKind::SendRejected
+            | AmqpErrorKind::AmqpDescribedError(_) => None,
+        }
+    }
+}
+
+impl std::fmt::Display for AmqpError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match &self.kind {
+            AmqpErrorKind::SimpleMessage(msg) => write!(f, "{}", msg),
+            AmqpErrorKind::AzureCore(e) => {
+                write!(f, "Azure Core Error: {}", e)
+            }
+            AmqpErrorKind::ManagementStatusCode(status_code, d) => {
+                if let Some(d) = d {
+                    write!(
+                        f,
+                        "Management API returned status code: {} ({})",
+                        status_code, d
+                    )
+                } else {
+                    write!(f, "Management API returned status code: {}", status_code,)
+                }
+            }
+            AmqpErrorKind::ConnectionDetachedByRemote(err) => {
+                write!(f, "Remote connection detached with error: {}", err)
+            }
+            AmqpErrorKind::LinkDetachedByRemote(err) => {
+                write!(f, "Remote link detached with error: {}", err)
+            }
+            AmqpErrorKind::SessionDetachedByRemote(err) => {
+                write!(f, "Remote session detached with error: {}", err)
+            }
+            AmqpErrorKind::LinkClosedByRemote(err) => {
+                write!(f, "Remote link closed with error: {}", err)
+            }
+            AmqpErrorKind::SessionClosedByRemote(err) => {
+                write!(f, "Remote session closed with error: {}", err)
+            }
+            AmqpErrorKind::ConnectionClosedByRemote(err) => {
+                write!(f, "Remote connection closed with error: {}", err)
+            }
+            AmqpErrorKind::DetachError(err) => {
+                write!(f, "AMQP Detach Error: {} ", err)
+            }
+            AmqpErrorKind::SendRejected => {
+                write!(f, "AMQP Send Rejected with no error information")
+            }
+            AmqpErrorKind::TransportImplementationError(s) => {
+                write!(f, "Transport Implementation Error: {}", s)
+            }
+            AmqpErrorKind::ConnectionDropped(s) => {
+                write!(f, "Connection dropped: {}", s)
+            }
+            AmqpErrorKind::FramingError(s) => {
+                write!(f, "Connection Framing error: {}", s)
+            }
+            AmqpErrorKind::IdleTimeoutElapsed(s) => {
```

Comment:
For some errors, we start the write with "AMQP ...", for some we don't. Which is fine, I am not directly proposing either remove or add. Just bringing it to your attention, in case you've been looking at this code for so long you don't see it.